src/client/pfs_file.go

```diff
@@ -338,9 +336,11 @@ func (c APIClient) WithCreateFileSetClient(cb func(ModifyFile) error) (resp *pfs
 		return nil, err
 	}
 	defer func() {
-		if retErr == nil {
-			resp, retErr = ctfsc.Close()
+		r, err := ctfsc.Close()
+		if retErr == nil && err == nil {
+			resp = r
 		}
+		retErr = multierr.Combine(retErr, errors.Wrap(err, "close"))
```

Comment:
Great fix! good read for sure. so here this can wrap `io.EOF `error when it occurs? Will the caller ever need to check for this error differently. You're right the standard lib says this should never be wrapped lol but since it's already inspected the glue should hold. The do have another error for the case [`ErrUnexpectedEOF` ](https://github.com/golang/go/blob/bf2ef26be3593d24487311576d85ec601185fbf4/src/io/io.go#L48)If the EOF occurs unexpectedly in a structured data stream. 

https://github.com/golang/go/blob/bf2ef26be3593d24487311576d85ec601185fbf4/src/io/io.go#L38